src/main/java/seedu/address/logic/commands/InviteCommand.java

```diff
@@ -0,0 +1,143 @@
+            + "Parameters: "
+            + "[" + PREFIX_PARTICIPANT + "PARTICIPANT]...\n"
+            + "Example: invite p/Ben p/David";
+
+    public static final String MESSAGE_RESULT = "%s\n%s";
+
+    public static final String MESSAGE_RESULT_NONE_SUCCESS = "%s";
+
+    public static final String MESSAGE_SUCCESS_INVITE = "Invited \"%s\" into the activity.";
+
+    public static final String MESSAGE_DUPLICATE_PERSON_IN_ACTIVITY =
+            "Unable to invite \"%s\" as he/she already exists in the activity.";
+
+    public static final String MESSAGE_NON_UNIQUE_SEARCH_RESULT =
+            "Unable to invite \"%s\" as he/she has no unique search result in the contacts.";
+
+    public static final String MESSAGE_NO_ONE_TO_INVITE =
+            "There is no one to invite.";
+
+    public static final String MESSAGE_NO_VIEWED_ACTIVITY = "There is no viewed activity currently.";
+    public static final String MESSAGE_DUPLICATE_ENTRY = "\"%s\" has duplicate entries, the first one will be added.";
+    public static final String MESSAGE_INVALID_PERSON = "Blank name encountered.";
+
+    private final List<String> peopleToInvite;
+
+    /**
+     * Creates an InviteCommand to invite the specified {@code Person(s)}
+     */
+    public InviteCommand(List<String> peopleToInvite) {
+        requireNonNull(peopleToInvite);
+        this.peopleToInvite = peopleToInvite;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (peopleToInvite.size() == 0) {
+            throw new CommandException(MESSAGE_NO_ONE_TO_INVITE);
+        }
+
+        if (model.getContext().getType() != ContextType.VIEW_ACTIVITY) {
+            throw new CommandException(MESSAGE_NO_VIEWED_ACTIVITY);
+        }
+
+        Activity activityToInviteTo = model.getContext().getActivity().get();
+        ArrayList<Integer> participantIds = activityToInviteTo.getParticipantIds();
+
+        List<String> keywords;
+        List<Person> findResult;
+        List<Integer> idsToInvite = new ArrayList<>();
+        StringBuilder warningMessage = new StringBuilder();
+        StringBuilder successMessage = new StringBuilder();
+
+        for (String name : peopleToInvite) {
+
+            if (name.trim().equals("")) {
+                warningMessage.append(MESSAGE_INVALID_PERSON + "\n");
+                continue;
+            }
+
+            keywords = Arrays.asList(name.split(" "));
+            NameContainsAllKeywordsPredicate predicate = new NameContainsAllKeywordsPredicate(keywords);
+
+            findResult = model.findPersonAll(predicate);
+            assert findResult != null : "List of people in contacts should not be null.";
+
+            if (findResult.size() != 1) {
+                String warning = String.format(MESSAGE_NON_UNIQUE_SEARCH_RESULT, name);
+                warningMessage.append(warning).append("\n");
+                continue;
+            }
```

Comment:
At some point we probably should abstract out this name searching logic from activity, invite, expense (and disinvite?) commands cause its currently being repeated a lot.

Also we need to fix the issue of names being impossible to access if they are a sub-string of other names so I guess that can be done together in one go